torchqdynamics/solver_options.py

```diff
@@ -1,11 +1,19 @@
 class SolverOption:
-    def __init__(self, *, verbose: bool = True):
+    def __init__(self, *, verbose: bool = True, save_states: bool = True):
+        """...
+
+        Args:
+            save_states (bool, optional): If `True`, the state is saved at every
+                time value. If `False`, only the final state is stored and returned.
+                Defaults to `True`.
+        """
         self.verbose = verbose
+        self.save_states = save_states
 
 
 class FixedStep(SolverOption):
-    def __init__(self, *, dt: float, verbose: bool = True):
-        super().__init__(verbose=verbose)
+    def __init__(self, *, dt: float, **kwargs):
+        super().__init__(**kwargs)
```

Comment:
I'm wondering whether we should be explicit with the arguments here? If people read the code, it's nice to see all the possible arguments to each SolverOption class directly in one place.

Reply:
This is an endless debate in the community haha, I redirect you to this StackOverlow post which I quite agree with > https://stackoverflow.com/questions/73997582/should-i-repeat-parent-class-init-arguments-in-the-child-classs-init-o. I think it actually makes the code easier to read in these inheritance schemes, and refactoring (renaming one of the base class argument or adding one) so much easier.